Replaces the standard HTML select with a custom gforge-dropdown

Guilherme-devcode opened the pull request:
Replaces the standard HTML select with a custom gforge-dropdown
component for items per page selection, enhancing UI consistency and
usability. Updates event handling and table stories to align with new
dropdown usage.

Improves maintainability by using DropdownOption model for dropdown
items instead of plain numbers.

Guilherme-devcode committed:
Refactors items per page dropdown in table
component

Replaces the standard HTML select with a custom gforge-dropdown
component for items per page selection, enhancing UI consistency and
usability. Updates event handling and table stories to align with new
dropdown usage.

Improves maintainability by using DropdownOption model for dropdown
items instead of plain numbers.

Guilherme-devcode committed:
Refactors TableComponent with improved inputs and
outputs

- Marks all component properties and methods as public for clarity and
consistency
- Replaces Portuguese comments and strings with English counterparts
- Introduces new properties for enhanced pagination and context menu
functionality
- Updates event emitters to improve data handling and user interaction
feedback

Guilherme-devcode commented:
@CodiumAI-Agent /review

Guilherme-devcode commented:
@CodiumAI-Agent /review
auto_approve
--pr_reviewer.extra_instructions="
In the `possible issues` section, emphasize the following:
- Is the code logic efficient?
- 
"
--pr_reviewer.inline_code_comments=true
-i
--pr_reviewer.require_score_review=true
--pr_reviewer.require_can_be_split_review=true
--pr_reviewer.require_soc2_ticket=true
--pr_reviewer.num_code_suggestions="0"

QodoAI-Agent commented:
Auto-approval option for PR-Agent is disabled. You can enable it via a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/docs/REVIEW.md#auto-approval-1)

Guilherme-devcode commented:
@CodiumAI-Agent /improve
--pr_code_suggestions.extra_instructions="
Emphasize the following:
- Does the code logic cover relevant edge cases?
- Is the code logic clear and easy to understand?
- Is the code logic efficient?
- 
"
--pr_code_suggestions.num_code_suggestions_per_chunk="4"

QodoAI-Agent commented:
## PR Code Suggestions ✨

<!-- d6acea4 -->

<table><thead><tr><td>Category</td><td align=left>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </td><td align=center>Score</td></tr><tbody><tr><td rowspan=3><strong>Possible issue</strong></td>
<td>



<details><summary>Add null-check for dropdown option</summary>

___


**Ensure that the <code>onItemsPerPageChange</code> method properly handles cases where the <code>option</code> <br>parameter is null or undefined to avoid runtime errors.**

[projects/g-forge/src/lib/components/table/table.component.ts [204-206]](https://github.com/Guilherme-devcode/g-forge/pull/3/files#diff-05b2925394d00839810a24e906c52f2af50605b56ebb98a892c436c80ab8ee2aR204-R206)

```diff
 public onItemsPerPageChange(option: DropdownOption): void {
+  if (!option) return;
   this.itemsPerPage = option.value;
   this.currentPage = 1;
 }
```
<details><summary>Suggestion importance[1-10]: 8</summary>

Why: Adding a null-check for the `option` parameter in the `onItemsPerPageChange` method is a valuable safeguard against potential runtime errors, especially in cases where the dropdown might emit an undefined or null value. This improves the robustness of the code.


</details></details></td><td align=center>8

</td></tr><tr><td>



<details><summary>Add validation for undefined field</summary>

___


**Ensure the <code>onSort</code> method handles cases where <code>column.field</code> is undefined to prevent <br>sorting errors.**

[projects/g-forge/src/lib/components/table/table.component.ts [165-169]](https://github.com/Guilherme-devcode/g-forge/pull/3/files#diff-05b2925394d00839810a24e906c52f2af50605b56ebb98a892c436c80ab8ee2aR165-R169)

```diff
 public onSort(column: { field: string; sortable?: boolean }): void {
-  if (!column.sortable) return;
+  if (!column.sortable || !column.field) return;
   const currentOrder =
     this.filteredData.length > 1 &&
     ...
 }
```
<details><summary>Suggestion importance[1-10]: 8</summary>

Why: Adding a check for `column.field` being undefined in the `onSort` method is a significant improvement. It prevents potential runtime errors during sorting operations, ensuring the method handles edge cases gracefully.


</details></details></td><td align=center>8

</td></tr><tr><td>



<details><summary>Add null-check for scroll target</summary>

___


**Add a safeguard in the <code>onScroll</code> method to ensure the <code>target</code> is not null or undefined <br>before accessing its properties to prevent potential runtime errors.**

[projects/g-forge/src/lib/components/table/table.component.ts [224-228]](https://github.com/Guilherme-devcode/g-forge/pull/3/files#diff-05b2925394d00839810a24e906c52f2af50605b56ebb98a892c436c80ab8ee2aR224-R228)

```diff
 public onScroll(event: Event): void {
   const target = event.target as HTMLElement;
-  if (target.scrollHeight - target.scrollTop === target.clientHeight) {
+  if (target && target.scrollHeight - target.scrollTop === target.clientHeight) {
     this.scrolled.emit();
   }
 }
```
<details><summary>Suggestion importance[1-10]: 7</summary>

Why: Adding a null-check for the `target` in the `onScroll` method ensures that the code does not throw runtime errors if the event's target is unexpectedly null or undefined. This is a reasonable precaution and enhances code reliability.


</details></details></td><td align=center>7

</td></tr><tr><td rowspan=1><strong>General</strong></td>
<td>



<details><summary>Add validation for empty data</summary>

___


**Validate that <code>this.filteredData</code> is not empty before calculating <code>totalPages</code> to avoid <br>unnecessary computations or errors.**

[projects/g-forge/src/lib/components/table/table.component.ts [240-243]](https://github.com/Guilherme-devcode/g-forge/pull/3/files#diff-05b2925394d00839810a24e906c52f2af50605b56ebb98a892c436c80ab8ee2aR240-R243)

```diff
 public get totalPages(): number[] {
+  if (!this.filteredData || this.filteredData.length === 0) return [];
   const total = this.filteredData.length;
   const pages = Math.ceil(total / this.itemsPerPage);
   return Array.from({ length: pages }, (_, i) => i + 1);
 }
```
<details><summary>Suggestion importance[1-10]: 6</summary>

Why: Adding a check for empty `filteredData` in the `totalPages` getter is a minor improvement. While it avoids unnecessary computations, the impact is limited since the existing code would still function correctly without this change.


</details></details></td><td align=center>6

</td></tr></tr></tbody></table>

Guilherme-devcode committed:
Enhances table component styling and interactivity

Updates table and action column styling for better visual hierarchy and
responsiveness.
Introduces hover states and transitions for improved user interaction.
Standardizes action buttons and adds visual feedback on hover.
Tidies up CSS rules and improves maintainability with clearer style
definitions.
